pkg/api/types.go

```diff
@@ -584,10 +584,28 @@ const (
 	ClaimLost PersistentVolumeClaimPhase = "Lost"
 )
 
+type HostPathType string
+
+const (
+	// If nothing exists at the given path, an empty directory will be created there. Otherwise, behaves like HostPathExists
+	HostPathUnset HostPathType = ""
+	// If nothing exists at the given path, the pod will fail to run and provide an informative error message
+	HostPathExists HostPathType = "exists"
+	// If a file does not exist at the given path, the pod will fail to run and provide an informative error message
```

Comment:
I do think we need the "or make one" semantic.  How about

Exists
File  // exists or create
FileExists  // must exist
Directory  // exists or create
DirectoryExists  // must exist
SocketExists
BlockDevExists
CharDevExists

or is it clearer to flip it:

Exists
File  // must exist
FileOrCreate
Directory  // must exist
DirectoryOrCreate
Socket  //must exist
BlockDev  //must exist
CharDev // must exist

I think I like the second better.  And I guess the default should be DirectoryOrCreate